src/wallet/rpcwallet.cpp

```diff
@@ -1141,13 +1168,39 @@ UniValue addmultisigaddress(const JSONRPCRequest& request)
     if (!request.params[2].isNull())
         strAccount = AccountFromValue(request.params[2]);
 
+    int required = request.params[0].get_int();
+
+    // Get the public keys
+    const UniValue& keys_or_addrs = request.params[1].get_array();
+    std::vector<CPubKey> pubkeys;
+    pubkeys.resize(keys_or_addrs.size());
+    for (unsigned int i = 0; i < keys_or_addrs.size(); ++i) {
+        if (IsHex(keys_or_addrs[i].get_str())) {
+            _createmultisig_hex_to_pubkey(pubkeys.at(i), keys_or_addrs[i].get_str());
+        } else {
+            _createmultisig_addr_to_pubkey(pwallet, pubkeys.at(i), keys_or_addrs[i].get_str());
+        }
+        if (!pubkeys.at(i).IsFullyValid()) {
```

Comment:
Same as in the non-wallet call - this is superfluous now.

Reply:
Done.